### PR TITLE
Geolocate /ingest endpoint

### PR DESCRIPTION
**Pull request by darkdarkdragon**
**What does this pull request do? Explain your changes. (required)**
For `/ingest` endpoint find closes server. Uses same code as `/broadcast` endpoint.

**Specific updates (required)**

- 
- 
- 

**How did you test each of these updates (required)**

**Does this pull request close any open issues?**

**Screenshots (optional):**

**Checklist:**

- [ ] I have read the **CONTRIBUTING** document.
- [ ] My change requires a change to the documentation.
- [ ] I have updated the documentation accordingly.
- [ ] I have added tests to cover my changes.

### CODE REVIEW

**Review by iameli — CHANGES_REQUESTED** [review]
So our transcoding servers are going to be located at `https://chi.livepeer.com` and `https://rkv.livepeer.com`. But our ingest servers are going to be located at `rtmp://mdw.livepeer.com` and `rtmp://fra.livepeer.com`. So we're going to need a new command-line option for these, perhaps `--ingest-region`?

**Comment by darkdarkdragon** [comment]
@iameli so there will own API service in the `mdw.livepeer.com` and own in the `fra.livepeer.com`?

File: packages/api/src/controllers/broadcaster.js [hunk 1]
```diff
@@ -15,11 +15,11 @@ export const getBroadcasterStatuses = async req => {
   return statuses
 }
 
-export const amalgamate = async req => {
+export const amalgamate = async (req, endpoint) => {
   const { servers } = req.region
   let responses = await Promise.all(
     servers.map(async ({ server }) => {
-      const serverRes = await fetch(`https://${server}/api/broadcaster`, {
+      const serverRes = await fetch(`http://${server}/api/${endpoint}`, {
```
Comment (iameli) [inline packages/api/src/controllers/broadcaster.js]: I assume you did this for testing, but they're https-only in production

File: packages/api/src/middleware/geolocate.js [hunk 2]
```diff
@@ -1,15 +1,14 @@
 'use strict'
 
-function geoLocateFactory(params) {
+function geoLocateFactory({ first = true, region = 'region' }) {
   return async (req, res, next) => {
-    let first = params.first || true
     if (req.query.first === 'false') {
       first = false
-    } else if ((req.query.first = 'true')) {
+    } else if ((req.query.first == 'true')) {
```
Comment (iameli) [inline packages/api/src/middleware/geolocate.js]: Whoa, good catch, we need some linting on here...

File: packages/api/src/controllers/broadcaster.js (hunk 1 above)
Comment (iameli) [reply packages/api/src/controllers/broadcaster.js]: Arguably we should specify these as `INGEST_REGION__0=https://mdw.livepeer.com` so we can allow both HTTP and HTTPS and ports and stuff. But it should at least default to `https` for backwards-compatibility.

File: packages/api/src/parse-cli.js [hunk 3]
```diff
@@ -133,11 +144,17 @@ export default function parseCli(argv) {
...
+            'hardcoded list of ingest points to return from /api/ingest.',
+          type: 'string',
+          default: '[]',
+        },
         supportAddr: {
           describe:
             'email address where outgoing emails originate. should be of the form name/email@example.com',
           type: 'string',
-          coerce: supportAddr => {
+          coerce: (supportAddr) => {
```
Comment (iameli) [inline packages/api/src/parse-cli.js:157]: It looks like your IDE has a different version of Prettier than we're using or something, there's a few of these format changes in this PR. Could you run this and see if it makes everything consistent?

```
git diff --name-only master | xargs npx prettier --write
```

**Review by iameli — CHANGES_REQUESTED** [review]
Solid, just polish

File: packages/api/src/controllers/broadcaster.js (hunk 1 above)
Comment (darkdarkdragon) [reply packages/api/src/controllers/broadcaster.js]: I've added ability to specify `--region` and `region-ingest` with or without protocol, defaulting to `https`

File: packages/api/src/middleware/geolocate.js (hunk 2 above)
Comment (darkdarkdragon) [reply packages/api/src/middleware/geolocate.js]: Yes, preferable it should be typescript compiler 😄

File: packages/api/src/parse-cli.js (hunk 3 above)
Comment (darkdarkdragon) [reply packages/api/src/parse-cli.js:157]: > Could you run this and see if it makes everything consistent?

and this added more parentheses around arrowed functions 😄 
I've looked into prettier docs - adding parentheses is default option, and we not overriding it anywhere - so why you have different results?

Comment (iameli) [reply packages/api/src/parse-cli.js:157]: Hah, great question, super weird. No problem then.

**Review by iameli — APPROVED** [review]
LGTM!

**Pull request merged**